src/main/java/com/google/impactdashboard/server/api_utilities/ResourceRetriever.java

```diff
@@ -75,6 +76,74 @@ protected ProjectListRetriever(CloudResourceManager cloudResourceManagerService)
     return projects;
   }
 
+  /**
+   * Returns the top level ancestor to a project which is the organization the project belongs under.
+   * @param projectId the project the organization id is being retrieved for.
+   */
+  public String getOrganizationId(String projectId){
+    List<Ancestor> ancestors;
+    try {
+       ancestors = getProjectAncestry(projectId);
+    } catch (IOException io) {
+      throw new RuntimeException("Failed to get ancestors: " + io.getMessage());
+    }
+    if(ancestors == null) {
+      return "N/A";
+    }
+    return ancestors.get(ancestors.size()-1).getResourceId().getId();
+  }
+
+  /**
+   * Returns the ancestor hierarchy for the specified project.
+   * @param projectId the project to get the ancestors for
+   */
+  private List<Ancestor> getProjectAncestry(String projectId) throws IOException {
+    CloudResourceManager.Projects.GetAncestry ancestry = cloudResourceManagerService.
+        projects().getAncestry(projectId, new GetAncestryRequest());
+    GetAncestryResponse response = ancestry.execute();
+    return response.getAncestor();
+  }
+
+  /**
+   * Returns the name of the organization that has the id specified.
+   * @param organizationId The id of the organization the name is retrieved for.
+   */
+  public String getOrganizationName(String organizationId) {
+    Organization organization;
+    try {
+      organization = searchOrganizationIds(organizationId);
+    } catch (IOException io) {
+      throw new RuntimeException("Could not find any organization with id:" + organizationId);
+    }
+    return organization != null ? organization.getDisplayName(): organizationId;
+  }
+
+  /**
+   * Searches through all the organizations to find the org that matches the id specified.
+   * @param organizationId The id of the organization that is being searched for.
+   * @return The organization that has the id specified, null if it does not exist.
+   */
+  private Organization searchOrganizationIds(String organizationId) throws IOException {
+    SearchOrganizationsRequest request = new SearchOrganizationsRequest();
+    CloudResourceManager.Organizations.Search search = cloudResourceManagerService.organizations()
+        .search(request);
```

Comment:
Since you already have the organization id, wouldn't get method be more efficient than search?

Reply:
The get method has weird documentation saying that it requires organization name rather than organization id. Apparently the the actual functionality requires the id. I will change this ASAP.